libs/community/langchain_community/document_loaders/parsers/pdf.py

```diff
@@ -119,16 +119,29 @@ def _extract_text_from_page(page: "PageObject") -> str:
                 )
 
         with blob.as_bytes_io() as pdf_file_obj:  # type: ignore[attr-defined]
-            pdf_reader = pypdf.PdfReader(pdf_file_obj, password=self.password)
+            import pypdf.errors
 
-            yield from [
-                Document(
-                    page_content=_extract_text_from_page(page=page)
-                    + self._extract_images_from_page(page),
-                    metadata={"source": blob.source, "page": page_number},  # type: ignore[attr-defined]
-                )
-                for page_number, page in enumerate(pdf_reader.pages)
-            ]
+            try:
+                pdf_reader = pypdf.PdfReader(pdf_file_obj, password=self.password)
+            except pypdf.errors.PyPdfError as e:
+                raise e
```

Comment:
This appears to have no effect as we raise the original error with no further action.